File: php/src/com/example/script/primary.php
<?php

// 必须使用相对于src目录的路径
require_once 'com/example/tool/ToolB.php';

// 使用 ToolB.php 中的 ToolB 类
use com\example\tool\ToolB;

/** 通过use使用java中的类, 不能使用 use xxx as xxx 只能使use,因为下面这句话会被替换为 import android.widget.Button, 这样写的目的是有代码提示 **/

// 声明全局变量, 如果不声明的话也可以但是没有代码提示
global $logger;
ToolB::methodB($logger);

/** @var m8test_java\com\m8test\script\core\api\ui\compose\ComposeView $composeView */
$composeView->create(function ($slot) {
    // 创建一个垂直布局
    $slot->Column(function ($column) {
        // 设置布局的水平对齐方式为水平居中
        $column->setHorizontalAlignment(function ($alignments) {
            return $alignments->getCenterHorizontally();
        });
        // 设置布局的垂直排列方式为居中
        $column->setVerticalArrangement(function ($arrangements) {
            return $arrangements->getCenter();
        });
        // 设置布局填满父容器
        $column->setModifier(function ($modifier) {
            return $modifier->fillMaxSize(1.0);
        });
        // 设置垂直布局的内容
        $column->setContent(function ($columnSlot) {
            // 1. 创建状态
            $state = $columnSlot->remember(function () use ($columnSlot) {
                return $columnSlot->mutableStateOf(0);
            });
            $columnSlot->Text(function ($text) use ($state) {
                // 2. 组合期间读取状态，状态变化时自动重组
                $text->setText("点击次数:" . $state->getValue());
            });
            $columnSlot->TextButton(function ($button) use ($state) {
                $button->setOnClick(function () use ($state) {
                    // 4. 点击按钮时， 更新状态的值
                    $state->setValue($state->getValue() + 1);
                });
                $button->setContent(function ($buttonSlot) {
                    $buttonSlot->Text(function ($text) {
                        $text->setText("点击");
                    });
                });
            });
        });
    });
});

// 启动界面
/** @var m8test_java\com\m8test\script\core\api\ui\Activity $activity */
$activity->start();

File: lua/src/com/example/script/primary.lua
-- 必须使用相对于src目录的路径
local ToolB = require("com.example.tool.ToolB")
ToolB.methodB(_G._logger)
_G._composeView:create(function(slot)
    slot:Column(function(column)
        -- 设置对齐方式为水平居中
        column:setHorizontalAlignment(function(alignments)
            return alignments:getCenterHorizontally()
        end)
        -- 设置排列方式为垂直居中
        column:setVerticalArrangement(function(columnArrangements)
            return columnArrangements:getCenter()
        end)
        -- 设置填充满整个父容器
        column:setModifier(function(modifier)
            return modifier:fillMaxSize(1.0)
        end)
        column:setContent(function(columnSlot)
            -- 1. 创建一个可变状态变量
            local state = columnSlot:remember(function()
                return columnSlot:mutableStateOf(0)
            end)
            columnSlot:Text(function(text)
                -- 2. 让文本内容跟踪状态变量,当状态变量变化时,文本内容会自动更新
                -- 3. 使用状态变量的值来设置文本内容
                text:setText("点击次数" .. state:getValue())
            end)
            columnSlot:TextButton(function(textButton)
                textButton:setContent(function(rowScopeSlot)
                    rowScopeSlot:Text(function(text)
                        text:setText("点击")
                    end)
                end)
                textButton:setOnClick(function()
                    -- 4. 点击按钮时,更新状态变量的值
                    state:setValue(state:getValue() + 1)
                end)
            end)
        end)
    end)
end)
_G._activity:start()
